CMakeLists.txt

```diff
@@ -255,8 +260,11 @@ install (FILES ${NLOPT_HEADERS} DESTINATION ${RELATIVE_INSTALL_INCLUDE_DIR})
 set (nlopt_lib nlopt)
 add_library (${nlopt_lib} ${NLOPT_SOURCES})
 add_dependencies(${nlopt_lib} generate-cpp)
+if(NLOPT_WITH_LUKSAN)
+  set(nlopt_lib_compile_definitions WITH_LUKSAN)
+endif()
+target_compile_definitions(${nlopt_lib} PUBLIC ${nlopt_lib_compile_definitions})
```

Comment:
seems nlopt_lib_compile_definitions is unused, you can directly use target_compile_definitions